src/visualizers/FactorHistogram.ts

```diff
@@ -101,28 +114,27 @@ class FactorHistogram extends P5GLVisualizer(paramDesc) {
             if (counter === 0 && math.bigabs(this.seq.getElement(i)) > 1) {
                 counter = -1
             }
-            if (counter in factorCount) factorCount[counter]++
-            else factorCount[counter] = 1
+            if (counter in this.factorCount) this.factorCount[counter]++
+            else this.factorCount[counter] = 1
         }
-        return factorCount
     }
 
     // Create an array with the frequency of each number
     // of factors in the corresponding bins
-    async binFactorArraySetup() {
+    async presketch(seqChanged: boolean, sizeChanged: boolean) {
+        this.precomputing = 'Evaluating sequence entries ...'
+        await super.presketch(seqChanged, sizeChanged)
+        if (!seqChanged) {
+            this.precomputing = ''
+            return
+        }
+        this.precomputing = 'Calculating sequence values ...'
+        await this.seq.fill(this.endIndex())
+        this.precomputing = 'Factoring ...'
```

Comment:
So sometimes it says "Factoring ...-1" on the screen.  For example, if you choose a random sequence and then choose a large upper bound on the random values, it will get into this state in Firefox for quite a while before it gets the histogram drawn.  What does the "-1" mean?